moon/controller_round1.py

```diff
@@ -13,17 +13,22 @@ def on_object_reached(self, timestamp, data):
         object_type = data
         x,y,z = self.xyz
         print(self.time, "app: Object %s reached" % object_type)
-        response = self.send_request('artf %s %f %f 0.0\n' % (object_type, x, y)).decode("ascii") 
-        print(self.time, "app: Volatile report response: %s" % response)
-        if response == 'ok':
-            pass
-        else:
-            # do nothing, ie keep going around and try to match the view
-            pass
+        def process_volatile_response(response):
+            print(self.time, "app: Volatile report response: %s" % response)
+            if response == 'ok':
+                pass
+            else:
+                # do nothing, ie keep going around and try to match the view
+                pass
+            
+        self.send_request('artf %s %f %f 0.0\n' % (object_type, x, y), process_volatile_response)
 
     def run(self):
-        message = self.send_request('request_origin').decode("ascii") 
-        print ("controller round 1: origin received: %s" % message)
+
+        def register_origin(message):
+            print ("controller round 1: origin received: %s" % message)
+
+        self.send_request('request_origin', register_origin)
```

Comment:
my biggest problem with this PR is "do we really need these bits asynchronously?"

Reply:
See my other comment/response, there were already problems as a result and it is an inferior design for robotics in general, likely to haunt us going forward.